Mactivision Mini-Games/Assets/Digger/Scripts/DiggerLevelManager.cs

```diff
@@ -11,33 +12,37 @@ public class DiggerLevelManager : LevelManager
     int digAmount; // total amount of presses required; must be > 0, rounds up to nearest 10
 
     List<KeyCode> keysDown; // List of keys currently held down (not full history)
-    InputRecorder recorder; // input recorder (this will record full history)
-    bool recording;
+    ButtonPressingMetric bpMetric; // records button pressing data during the game
+    MetricJSONWriter metricWriter; // outputs recording metric (bpMetric) as a json file
 
     // Start is called before the first frame update
     void Start()
     {
         Setup(); // run initial setup, inherited from parent class
         countDoneText = "Dig!";
         digKey = KeyCode.B;
-        digAmount = 10;
+        digAmount = 100;
         keysDown = new List<KeyCode>();
-        recorder = new InputRecorder();
-        recording = false;
+        bpMetric = new ButtonPressingMetric();
+        metricWriter = new MetricJSONWriter("Digger", DateTime.Now);
     }
 
     // Update is called once per frame
     void Update()
     {
         if (lvlState==2) {
-            if (!recording) { // begin recording 
-                recording = true;
-                recorder.StartRec();
+            if (!bpMetric.isRecording) { // begin recording 
+                bpMetric.startRecording();
                 SetDigKeyForGround(); // not called at Start() because this script could load before the blocks
                 SetDigAmountForGround();
             }
             if (chest.opened) { // the player landing on chest triggers the end of the game
-                recorder.EndRec();
+                bpMetric.finishRecording();
+                metricWriter.logMetrics(
+                    "Logs/digger_"+DateTime.Now.ToFileTime()+".json", 
```

Comment:
I think `DateTime.Now` should be put in a variable prior to this line so that we don't get slightly different times for the filename and the time saved in the file.